### PR TITLE
[VideoPlayer] FFmpegDemux: proper flush io context.

### PR DESCRIPTION
**Pull request by afedchin**
see title.

@FernetMenta ping

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp [hunk 1]
```diff
@@ -581,6 +581,9 @@ void CDVDDemuxFFmpeg::Reset()
 
 void CDVDDemuxFFmpeg::Flush()
 {
+  if (m_ioContext)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: test m_pFormatContext->pb in the block below. we don't create the ioContext for all cases ourselves.

Comment (afedchin) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: do you suggest to flush internal ffmpeg io?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: there is only one io context. either we create it an assign it to m_pFormatContext->pb or ffmpeg creates it. 
see https://github.com/afedchin/xbmc/blob/6596aa1901dcde54f7dfe6efba3de94940466cfb/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L385

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: also: https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/avformat.h#L2338

Comment (afedchin) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp]: updated

**Comment by FernetMenta** [comment]
thanks, looks good now

**Comment by afedchin** [comment]
now it better

**Comment by FernetMenta** [comment]
but the comment 2 lines above is outdated and wrong. those are no internal function anymore

**Comment by afedchin** [comment]
should we just drop the comment?

**Comment by FernetMenta** [comment]
yes please

**Comment by afedchin** [comment]
jenkins build and merge

**Pull request merged**